src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Providers/IaasVmPsBackupProvider.cs

```diff
@@ -280,6 +285,121 @@ public RecoveryPointBase GetRecoveryPointDetails()
+            string protectedItemName = HelperUtils.GetProtectedItemUri(uriDict, rp.Id);
+
+            IaasVMILRRegistrationRequest registrationRequest =
+                new IaasVMILRRegistrationRequest();
+            registrationRequest.RecoveryPointId = rp.RecoveryPointId;
+            registrationRequest.VirtualMachineId = rp.SourceResourceId;
+            registrationRequest.RenewExistingRegistration = (rp.IlrSessionActive == false) ? false : true;
+
+            var ilRResponse = ServiceClientAdapter.ProvisioninItemLevelRecoveryAccess(
+                containerUri, protectedItemName, rp.RecoveryPointId, registrationRequest);
+
+            IEnumerable<string> ie =
+                    ilRResponse.Response.Headers.GetValues("Azure-AsyncOperation");
+            string asyncHeader = string.Empty;
+            foreach (string s in ie)
+            {
+                asyncHeader = s;
+            }
+
+            AzureVmRPMountScriptDetails result = null;
+            var response = TrackingHelpers.GetOperationStatus(
+                ilRResponse,
+                operationId => ServiceClientAdapter.GetProtectedItemOperationStatus(operationId));
+
+            if (response != null && response.Status != null &&
+                   response.Properties != null && ((OperationStatusProvisionILRExtendedInfo)
+                   response.Properties).RecoveryTarget != null)
+            {
+                InstantItemRecoveryTarget recoveryTarget =
+                    ((OperationStatusProvisionILRExtendedInfo)
+                    response.Properties).RecoveryTarget;
+
+                if (recoveryTarget.ClientScripts.Count != 0)
+                {
+                    if (recoveryTarget.ClientScripts.Count == 2)
+                    {
+                        // clientScriptForConnection.OsType == "Windows"
+                        result = this.GenerateILRResponseForWindowsVMs(
+                                recoveryTarget.ClientScripts[1], out content);
+                    }
+                    else
+                    {
+                        // clientScriptForConnection.OsType == "Linux"
+                        result = this.GenerateILRResponseForLinuxVMs(
+                                recoveryTarget.ClientScripts[0],
+                                protectedItemName, rp.RecoveryPointTime.ToString(), out content);
+                    }
+                }
+            }
+
+            string scriptDownloadLocation =
+                    (string)ProviderData[RecoveryPointParams.FileDownloadLocation];
+            string absoluteFilePath = string.Empty;
+            if (string.IsNullOrEmpty(scriptDownloadLocation))
+            {
+                scriptDownloadLocation = Directory.GetCurrentDirectory();
+            }
+            absoluteFilePath = Path.Combine(scriptDownloadLocation, result.Filename);
+            File.WriteAllBytes(absoluteFilePath, Convert.FromBase64String(content));
```

Comment:
@dragonfly91 also, please use `AzureSession.Instance.DataStore` to read from/write to a file.